rero_mef/api_mef.py

```diff
@@ -106,58 +96,79 @@ def get_all_pids_without_viaf(cls):
             yield hit.pid
 
     @classmethod
-    def get_pids_with_multiple_mef(cls, record_types=None, verbose=False):
+    def get_pids_with_multiple_mef(cls, record_types=None, verbose=False,
+                                   before=1):
         """Get agent pids with multiple MEF records.
 
         :params record_types: Record types (pid_types).
         :param verbose: Verbose.
+        :param before: Get multiple MEF before x minutes (default 1 minute).
         :returns: pids, multiple pids, missing pids.
         """
         pids = {}
         multiple_pids = {}
         missing_pids = {}
-        if record_types:
-            for record_type in record_types:
-                if verbose:
-                    click.echo(f'Calculating {record_type}:')
+        none_pids = {}
+        agents = {}
+        if not record_types:
+            record_types = []
+        for record_type in record_types:
+            try:
+                agent_class = get_entity_class(record_type)
+                agents[record_type] = {
+                    'name': agent_class.name,
+                    'search': get_entity_search_class(record_type)(),
+                }
                 pids[record_type] = {}
                 multiple_pids[record_type] = {}
                 missing_pids[record_type] = []
-                if agent_class := get_entity_class(record_type):
-                    agent_name = agent_class.name
-                    query = cls.search().filter('exists', field=agent_name)
+                none_pids[record_type] = []
+            except Exception:
+                current_app.logger.error(
+                    f'Record type not found: {record_type}')
+
+            click.echo('Get mef')
+            date = datetime.utcnow() - timedelta(minutes=before)
+            progress = progressbar(
+                items=cls.search()
+                .filter('range', _created={'lte': date})
+                .params(preserve_order=True)
+                .sort({'_updated': {'order': 'desc'}})
+                .scan(),
+                length=cls.search().count(),
+                verbose=verbose
+            )
+            for hit in progress:
+                data = hit.to_dict()
+                mef_pid = data['pid']
+                for record_type, info in agents.items():
+                    if data.get(info['name'], {}) is None:
+                        none_pids[record_type].append(mef_pid)
+                    elif agent_pid := data.get(info['name'], {}).get('pid'):
+                        pids[record_type].setdefault(agent_pid, [])
+                        pids[record_type][agent_pid].append(mef_pid)
+                        if len(pids[record_type][agent_pid]) > 1:
```

Comment:
only `if pids[record_type][agent_pid]:` is possible

Reply:
We test for more then one.